tests/test_freqtradebot.py

```diff
@@ -1459,7 +1462,7 @@ def test_tsl_on_exchange_compatible_with_edge(mocker, edge_conf, fee, caplog,
     cancel_order_mock = MagicMock()
     stoploss_order_mock = MagicMock()
     mocker.patch('freqtrade.exchange.Exchange.cancel_order', cancel_order_mock)
-    mocker.patch('freqtrade.exchange.Exchange.stoploss', stoploss_order_mock)
+    mocker.patch('freqtrade.exchange.Binance.stoploss', stoploss_order_mock)
```

Comment:
Should this test be parametrized for the exchanges with stoploss on exchange implemented (Binance, Kraken)?

Reply:
i don't think so... 
this does not test the exchange-specific stoploss code - but the handling of stoploss within freqtradebot - which is identical for both exchanges.
Since the stoploss-method is mocked away, there can't be a difference in behaviour this test would be able to catch.

There is another way to fix this test - by mocking `stoploss_adjust()` too and keeping this test on the default exchange class (as it was before) - which basically calls 2 methods that are not implemented (throws OperationalException when not mocked).

Now since `stoploss_adjust()` has no side-effects and is just a small helper method, i didn't see the need to mock that, too.